### DISCUSSION TITLE
Resolver and rules

### DISCUSSION THREAD

**Discussion by Daneng66** [react-hook-form/react-hook-form#7512] [Q&A] [answered]
Can you use a combination of resolver and controller rules at the same time?

**Comment by Moshyfawn** [comment, marked as answer]
There's a section on resolvers [on this page](https://react-hook-form.com/api/useform/#validationResolver) of RHF docs.
I haven't done the setup you've mentioned with the resolver and controller rules combined myself in favor of the resolver schema to be the single source of truth for the validation. I can imaging it can get messy.
The RHF docs have this line about resolvers and the built-in validators: 
> "A resolver cannot be used with the built-in validators (e.g.: required, min, etc.)"

**Comment by Daneng66** [comment]
@Moshyfawn I was literally just commenting that I just found in the documentation 

> A resolver cannot be used with the built-in validators (e.g.: required, min, etc.)

Follow up question: What is the behaviour if your validation schema contains information for a field that is not registered? Will it still pass?

**Reply by Moshyfawn** [reply]
I guess, it depends on the validation schema.
Here, I've created [a simple codesandbox](https://codesandbox.io/s/blazing-tdd-xfwxv?file=/src/App.tsx) with Zod validation library, where the second input hasn't been registered. The schema implies that the number of characters for each field is more than 1, so the validation will fail regardless if the input is registered or not. If you mark a field as optional _(`.optional()` for Zod, since it's required first)_, you'll be able to submit the form, since.. well, the form value passes the validation that way
